docs/best-practices/crate-node.txt

```diff
@@ -1,434 +1,442 @@
-.. highlight:: psql
-.. _recover:
+.. highlight:: console
+.. _crate-node-cli:
 
-====================================
-Troubleshooting with Crate-Node CLI
-====================================
+============================================
+Troubleshooting With The Crate-Node CLI Tool
+============================================
 
-The crate-node CLI enables you to perform certain unsafe operations
-on a node that are only possible while it is shut down. This command allows you
-to adjust the role of a node and may be able to recover some data after a
-disaster.
+The crate-node CLI tool allows you execute certain troubleshooting operations
+on nodes and CrateDB clusters. These operations include forming a new CrateDB
+cluster, repairing a broken cluster, broken nodes.
 
-If a CrateDB cluster unrecoverably looses the majority of the master eligible
-nodes, the crate-node CLI provides a way to utilize the state of the remaining
-master nodes to form a new cluster.
+In this section, you learn how to change the states of remaining master nodes to
+form a new cluster if the majority of the master-eligible nodes are
+irrecoverably lost. Changing the cluster states of nodes includes certain unsafe
+operations that you can perform only on nodes that are shut down.
 
-Nodes may be temporarily stopped for maintenance reasons or systems
-failures. Once the issues on the nodes are resolved, a restart is necessary
-to let them join the cluster again. Unfortunately this is not always
-possible. A node might be unrepairable and does not start-up,
-so if the cluster is still available the node can be simply replaced by
-a fresh node on a new host which will replace the broken node.
+.. rubric:: Table of contents
 
-crate-node CLI
-==============
+.. toctree::
+   :maxdepth: 1
 
-.. code-block:: console
+.. contents::
+   :local:
 
-   bin/crate-node repurpose|unsafe-bootstrap|detach-cluster
-   [--ordinal <Integer>] [-C <KeyValuePair>]
-   [-h, --help] ([-s, --silent] | [-v, --verbose])
+Changing the role of a node – ``crate-node`` repurpose
+------------------------------------------------------
 
+.. NOTE::
 
-The following commands provide a various operations to modify and recover nodes:
+  You can only perform the following operations on an affected node after you
+  have shut it down.
 
-* ``crate-node repurpose`` can be used to delete unwanted data from a node if it
-  used to be a data node or a master-eligible node but has been repurposed not to
-  have one or other of these roles.
+The command ``crate-node repurpose`` is intended to:
```

Comment:
A) This sentence: —The command ``crate-node repurpose`` is intended to:— was misleading due to poor rewriting. I’ve corrected that. We instead outline the overall procedure of changing the role of a node at this point.

B) Step three of the description of the procedure shows that the command is run on a node.

C) Taking that into account I implemented link request and amended the prose so that the reader gets this bit's meaning right.

D) I rephrased both the paras below and included proper examples of the command in use.

*Source-content:*

Changing the role of a node

There may be situations where you want to repurpose a node without following the proper repurposing processes. The elasticsearch-node repurpose tool allows you to delete any excess on-disk data and start a node after repurposing it.

The intended use is:

Stop the node
Update elasticsearch.yml by setting node.master and node.data as desired.
Run elasticsearch-node repurpose on the node
Start the node

If you run elasticsearch-node repurpose on a node with node.data: false and node.master: true then it will delete any remaining shard data on that node, but it will leave the index and cluster metadata alone. If you run elasticsearch-node repurpose on a node with node.data: false and node.master: false then it will delete any remaining shard data and index metadata, but it will leave the cluster metadata alone.